vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthCaptchaFragment.kt

```diff
@@ -40,22 +47,71 @@ class FtueAuthCaptchaFragment @Inject constructor(
 ) : AbstractFtueAuthFragment<FragmentFtueLoginCaptchaBinding>() {
 
     private val params: FtueAuthCaptchaFragmentArgument by args()
+    private var webViewBinding: ViewStubWebviewBinding? = null
     private var isWebViewLoaded = false
 
     override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentFtueLoginCaptchaBinding {
-        return FragmentFtueLoginCaptchaBinding.inflate(inflater, container, false)
+        return FragmentFtueLoginCaptchaBinding.inflate(inflater, container, false).also {
+            it.loginCaptchaWebViewStub.setOnInflateListener { _, inflated ->
+                webViewBinding = ViewStubWebviewBinding.bind(inflated)
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        inflateWebViewOrShowError()
+    }
+
+    private fun inflateWebViewOrShowError() {
+        views.loginCaptchaWebViewStub.inflateWebView(onError = {
+            MaterialAlertDialogBuilder(requireActivity())
+                    .setTitle(R.string.dialog_title_error)
+                    .setMessage(it.localizedMessage)
+                    .setPositiveButton(R.string.ok) { _, _ ->
+                        requireActivity().recreate()
+                    }
+                    .show()
+        })
     }
 
     override fun resetViewModel() {
         viewModel.handle(OnboardingAction.ResetAuthenticationAttempt)
     }
 
     override fun updateWithState(state: OnboardingViewState) {
-        if (!isWebViewLoaded) {
-            captchaWebview.setupWebView(this, views.loginCaptchaWevView, views.loginCaptchaProgress, params.siteKey, state) {
+        if (!isWebViewLoaded && webViewBinding != null) {
+            captchaWebview.setupWebView(this, webViewBinding!!.root, views.loginCaptchaProgress, params.siteKey, state) {
                 viewModel.handle(OnboardingAction.PostRegisterAction(RegisterAction.CaptchaDone(it)))
             }
             isWebViewLoaded = true
         }
     }
 }
+
+private fun ViewStub.inflateWebView(onError: (Throwable) -> Unit) {
+    try {
+        inflate()
+    } catch (e: Exception) {
+        val isMissingWebView = e.crawlCausesFor { it.message?.contains("MissingWebViewPackageException").orFalse() }
+        if (isMissingWebView) {
+            onError(MissingWebViewException(e))
+        } else {
+            onError(e)
+        }
+    }
+}
+
+private fun Throwable?.crawlCausesFor(predicate: (Throwable) -> Boolean): Boolean {
+    return when {
+        this == null -> false
+        else -> {
+            when (predicate(this)) {
+                true -> true
+                else -> this.cause.crawlCausesFor(predicate)
+            }
+        }
+    }
+}
```

Comment:
Could be rewritten to this compact, but maybe less clear:

```kotlin
private fun Throwable.crawlCausesFor(predicate: (Throwable) -> Boolean): Boolean {
    return predicate(this) || cause?.crawlCausesFor(predicate) ?: false
}
```

Also, maybe move this extension to a dedicated file for Throwable extensions?

Reply:
will extract :+1: 

as the function is recursive I'd lean towards making the exit conditions as clear as possible, having the recursive call the last line allows us to opt into the tailrec optimisations (although it hasn't been used here since the crash stack is fairly small)